### PR TITLE
UHF-7111 Unified card style

### PR DESCRIPTION
**Pull request by Arkkimaagi**
# [UHF-7111](https://helsinkisolutionoffice.atlassian.net/browse/UHF-7111)

Unified card style for react district and project search

## What was done

* Created new reusable components
  * Card
  * Icon
  * Tags

## How to install

* You know better

## How to test

* [ ] Check that this feature works
* [ ] Check that code follows our standards

## Designers review

* This PR does not need designers review

## Other PRs

* https://github.com/City-of-Helsinki/drupal-hdbt/pull/489

### CODE REVIEW

File: src/components/common/Icon.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+interface IconProps
```
Comment (Jussiles) [inline src/components/common/Icon.tsx:1]: I would write this like this: 
```
interface IconProps {
  icon: string;
  className?: string;
  label?: string;
};
```

**Review by Jussiles — APPROVED** [review]
Works well and looks good! 🦜  mostly added some nitpick comments about coding style.

File: src/components/common/Icon.tsx (hunk 1 above)
Comment (Arkkimaagi) [reply src/components/common/Icon.tsx:1]: Fixed

File: src/components/common/Tags.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import TagType from '../../types/TagType';
+
+interface TagsProps
+{
```
[the remark this answers is not in the document]
Comment (Arkkimaagi) [reply src/components/common/Tags.tsx:4]: Fixed

File: src/components/common/Tags.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  isInteractive?: boolean;
+};
+
+export const Tags = (props: TagsProps): JSX.Element => {
+  const {
+    tags,
+    isInteractive,
+  } = props;
+
+  const typeClass = isInteractive ?
```
[the remark this answers is not in the document]
Comment (Arkkimaagi) [reply src/components/common/Tags.tsx:15]: Fixed

File: src/components/common/Tags.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    'content-tags__tags--static';
+
+  return (
+    <section
+      className="content-tags"
+      aria-label={
+        Drupal.t('Tags', {}, { context: 'Label for screen reader software users explaining that this is a list of tags related to this page.' })
+      }>
+      <ul className={`content-tags__tags ${typeClass}`}>
+        {tags.map((item:any, key:number) =>
```
[the remark this answers is not in the document]
Comment (Arkkimaagi) [reply src/components/common/Tags.tsx:27]: Yes, Fixed

File: src/components/common/Card.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+          { cardImage }
+        </div>
+      }
+
+      <div className="card__text">
+        <HeadingTag className="card__title">
+          <a href={cardUrl} className="card__link" {...cardUrlExternal && { 'data-is-external': 'true' }} rel="bookmark">
+            <span>{ cardTitle }</span>
+            {cardUrlExternal &&
+              <span className="link__type link__type--external" aria-label={`(${Drupal.t('Link leads to external service', {}, { context: 'Explanation for screen-reader software that the icon visible next to this link means that the link leads to an external service.' })})`}></span>
```
[the remark this answers is not in the document]
Comment (Arkkimaagi) [reply src/components/common/Card.tsx:58]: Yep, Fixed

File: src/components/common/Card.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+                <span className="card__meta__label">{cardMeta.label}: </span>
+                <span className="card__meta__content">{cardMeta.content}</span>
+              </div>
+             )}
+          </div>
+        }
+
+        {cardTags &&
+          <div className="card__tags">
+            <Tags tags={ cardTags } />
```
[the remark this answers is not in the document]
Comment (Arkkimaagi) [reply src/components/common/Card.tsx:102]: Fixed

File: src/components/results/ResultCard.tsx [hunk 7]
```diff
@@ -17,63 +20,112 @@ const ResultCard = ({
...
-      <div className={`${content_type[0]}-teaser__image`}>
-        { imageUrl ? (
-          <img src={`${imageUrl}`} alt={imageAlt} loading="lazy" typeof="foaf:Image" />
-        ) : (
-        <div className="image-placeholder">
-          <span className="hel-icon hel-icon--home-smoke"></span>
-        </div>
-        )}
-      </div>
+  if (imageAlt == '""') {
```
[the remark this answers is not in the document]
Comment (Arkkimaagi) [reply src/components/results/ResultCard.tsx:29]: Indeed, Fixed

File: src/components/results/ResultCard.tsx [hunk 8]
```diff
@@ -17,63 +20,112 @@ const ResultCard = ({
...
+  const getTimeItem = (dateStrings: any): JSX.Element => (
+    dateStrings.map((dateString: number, i: number) => (
+      <time dateTime={getHtmlTime(dateString)} key={`${dateString}-${i}`}> {i !== 0 && "-"} {getVisibleTime(dateString)}</time>
+    ))
+  );
+
+  const metas:Array<MetadataType> = [];
+
+  if (project_plan_schedule || project_execution_schedule) {
+    const schedule: any = (
```
[the remark this answers is not in the document]
Comment (Arkkimaagi) [reply src/components/results/ResultCard.tsx:70]: Seems to work, Fixed

File: src/components/results/ResultCard.tsx [hunk 9]
```diff
@@ -17,63 +20,112 @@ const ResultCard = ({
...
+    metas.push({
+      icon: 'locate',
+      label: Drupal.t('Theme'),
+      content: field_project_theme_name,
+    })
+  }
+
+  return (
+    <Card
+      cardAbout={'test'}
```
[the remark this answers is not in the document]
Comment (Arkkimaagi) [reply src/components/results/ResultCard.tsx:120]: After discussion, removed all references to this drupalism. Fixed

**Pull request merged**